server/src/main/java/org/opensearch/search/sandboxing/TaskData.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.sandboxing;
+
+import java.util.List;
+import java.util.Map;
+
+import org.opensearch.tasks.Task;
+
+public class TaskData {
```

Comment:
I think we should move the data from **this class** and `ResourceUsageData` into `CancellationContext` instead of having multiple layers of indirection. SInce we are adding this code in core hence should minimise adding unnecessary construct.